File: go/micro-blog/micro_blog.go
// Micro Blog exercise.
package microblog

// Truncate a UTF-8 string to five characters (or less)
// @param phrase: The string to truncate
// @returns: New string containing up to the first five characters of the given phrase
func Truncate(phrase string) string {
	const MAX_COUNT int = 5
	runes := []rune(phrase)
	count_runes := min(MAX_COUNT, len(runes))
	return string(runes[:count_runes])
}

File: php/micro-blog/MicroBlog.php
<?php
// Micro-Blog Exercise. Trim to 5 code points

declare(strict_types=1);

// MicroBlog class holds the truncate function.
class MicroBlog
{
    // Truncate a multi-byte unicode string to five characters.
    // @param $text: The text to split.
    // @returns: String of up to five runes/code points.
    public function truncate(string $text): string
    {
        return mb_substr($text, 0, 5);
    }
}
